tool/lib/commands/serve.dart

```diff
@@ -0,0 +1,137 @@
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:devtools_tool/model.dart';
+import 'package:io/io.dart';
+import 'package:path/path.dart' as path;
+
+import '../utils.dart';
+
+const _useLocalFlutterFlag = 'use-local-flutter';
+const _updatePerfettoFlag = 'update-perfetto';
+const _buildAppFlag = 'build-app';
+
+/// This command builds DevTools in release mode by running the
+/// `devtools_tool build-release` command and then serves DevTools with a
+/// locally running DevTools server.
+///
+/// If the [_buildAppFlag] is negated (e.g. --no-build-app), then the DevTools
+/// web app will not be rebuilt before serving.
+///
+/// If [_useLocalFlutterFlag] is present, the Flutter SDK will not be updated to
+/// the latest Flutter candidate. Use this flag to save the cost of updating the
+/// Flutter SDK when you already have the proper SDK checked out.
+///
+/// If [_updatePerfettoFlag] is present, the precompiled bits for Perfetto will
+/// be updated from the `devtools_tool update-perfetto` command as part of the
+/// DevTools build process (e.g. running `devtools_tool build-release`).
+class ServeCommand extends Command {
+  ServeCommand() {
+    argParser
+      ..addFlag(
+        _useLocalFlutterFlag,
+        negatable: false,
+        defaultsTo: false,
+        help:
+            'Whether to use the Flutter SDK on PATH instead of the Flutter SDK '
+            'contained in the "tool/flutter-sdk" directory.',
+      )
+      ..addFlag(
+        _updatePerfettoFlag,
+        negatable: false,
+        defaultsTo: false,
+        help: 'Whether to update the Perfetto assets before building DevTools.',
+      )
+      ..addFlag(
+        _buildAppFlag,
+        negatable: true,
+        defaultsTo: true,
+        help:
+            'Whether to build the DevTools app in release mode before starting '
+            'the DevTools server.',
+      );
+  }
+
+  @override
+  String get name => 'serve';
+
+  @override
+  String get description =>
+      'Builds DevTools in release mode and serves the web app with a locally '
+      'running DevTools server.';
+
+  @override
+  Future run() async {
+    final repo = DevToolsRepo.getInstance();
+    final processManager = ProcessManager();
+
+    final useLocalFlutter = argResults![_useLocalFlutterFlag];
+    final updatePerfetto = argResults![_updatePerfettoFlag];
+    final buildApp = argResults![_buildAppFlag];
+
+    final remainingArguments = List.of(argResults!.arguments)
+      ..remove(_useLocalFlutterFlag)
+      ..remove(_updatePerfettoFlag)
+      ..remove(_buildAppFlag);
+
+    final localDartSdkLocation = Platform.environment['LOCAL_DART_SDK'];
+    if (localDartSdkLocation == null) {
+      throw Exception('LOCAL_DART_SDK environment variable not set. Please add '
+          'the following to your \'.bash_profile\' or \'.bashrc\' file:\n'
+          'export LOCAL_DART_SDK=<absolute/path/to/my/dart/sdk>');
+    }
```

Comment:
@DanTup how can we tweak this exception to be useful for windows? We can do 
```
final  exceptionMessage =  Platform.isWindows ? ... : ...;
throw Exception(exceptionMessage);
```

Reply:
I would probably just say something like

> Please set the LOCAL_DART_SDK environment variable in Windows to <absolute/path/to/my/dart/sdk>

Unfortunately I've never found a good page on the MS site on how to set environment variables, but there are plenty of unofficial sites with instructions for all versions of Windows.